### PR TITLE
Agile data

### PR DESCRIPTION
**Pull request by craigmulligan**

### CODE REVIEW

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -73,6 +73,16 @@ var agile = require('agile-sdk')({
...
             * [.authenticateClient(client, secret)](#agile.idm.authentication.authenticateClient) ⇒ <code>Promise</code>
+    * [.data](#agile.data) : <code>object</code>
+        * [.subscription](#agile.data.subscription) : <code>object</code>
+            * [.create(deviceID, componentID, [interval])](#agile.data.subscription.create) ⇒ <code>Promise</code>
+            * [.delete(deviceID, componentID)](#agile.data.subscription.delete) ⇒ <code>Promise</code>
+            * [.update(deviceID, componentID, [interval])](#agile.data.subscription.update) ⇒ <code>Promise</code>
+            * [.get([deviceID], [componentID])](#agile.data.subscription.get) ⇒ <code>Promise</code>
+        * [.record](#agile.data.record) : <code>object</code>
+            * [.get([query])](#agile.data.record.get) ⇒ <code>Promise</code>
+        * [.record](#agile.data.record) : <code>object</code>
```
Comment (craigmulligan) [inline DOCUMENTATION.md]: duplicate of above.

**Review by nopbyte — APPROVED** [review]
@craig-mulligan, 
The part that I can judge best would be the src/index.js and I don't see any conflicting changes with what I did before. So, it looks god. I skimmed through the rest, and with my limited knowledge of the agile-sdk code it looks OK.

File: src/data/retention.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    * @name update
+    * @public
+    * @function
+    * @memberof agile.data.retention
+    * @param retentionInterval {String} - retentionInterval
+    * @fulfil {Array}
+    * @returns {Promise}
+    *
+    * @example
+    * agile.data.retention.get()
```
Comment (craig-mulligan) [inline src/data/retention.js]: wrong method

File: src/data/index.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    * @memberof agile.data
+    **/
+    subscription: subscription(base),
+    /**
+    * @namespace record
+    * @memberof agile.data
+    **/
+    record: record(base),
+    /**
+    * @namespace record
```
Comment (craigmulligan) [inline src/data/index.js]: should be `retention`

File: src/data/record.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+const subscription = (base) => {
+  base = `${base}/record`;
+  return ({
+    /**
+    * @summary Get records from gateway
+    * @name get
+    * @public
+    * @function
+    * @memberof agile.data.record
+    * @param [query] {Object} - Basic query that is transformed to influx sql
```
Comment (craigmulligan) [inline src/data/record.js]: {String}

File: src/data/record.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    * @returns {Promise}
+    *
+    * @example
+    * agile.data.record.get()
+    * .then(function(subscription) {
+    *   console.log(subscription);
+    * });
+    *
+    * const query = 'where={"deviceID":"mySensor"}&order={"by":"time","direction":"ASC"}'
+    * agile.data.record.get({
```
Comment (craigmulligan) [inline src/data/record.js]: example is wrong

**Pull request merged**